components/proxy/src/main/java/com/hotels/styx/metrics/reporting/graphite/GraphiteReporter.java

```diff
@@ -230,12 +226,20 @@ public void report(SortedMap<String, Gauge> gauges,
             graphite.flush();
         } catch (Exception e) {
             LOGGER.error("Error reporting metrics" + e.getMessage(), e);
-            reconnectIfNecessary(e);
+            try {
+                graphite.close();
+            } catch (IOException e1) {
+                LOGGER.warn("Error closing Graphite", graphite, e1);
+            }
         }
     }
 
     private <M extends Metric> void doReport(String name, M metric, long timestamp, MetricReportingAction<M> consumer) {
         try {
+            if (!graphite.isConnected()) {
+                graphite.connect();
+            }
+
             consumer.execute(name, metric, timestamp);
         } catch (Exception e) {
             LOGGER.error("Error reporting metric '" + name + "': " + e.getMessage(), e);
```

Comment:
Hi @alobodzki could you provide some background for this change? Mainly for the record please.

Reply:
Our implementation of GraphiteReport tries to reconnect between failures in a single upload, but the implementation of reconnectIfNecessary reconnects only with IOException, there is no default option in that method, so theoretically we could enter doReport method without connected GraphiteConsole. Dropwizard implementation doesn't have this issue, as they don't try to recover, they just fail fast.